File: item-list/src/main/java/com/gurunars/item_list/EmptyViewBinder.java
package com.gurunars.item_list;

import android.content.Context;
import android.view.View;

/**
 * Renderer for a situation when the list has not items.
 */
public interface EmptyViewBinder {
    /**
     * Return an inflated view.
     *
     * @param context to be used for rendering
     * @return rendered view
     */
    View getView(Context context);

}

File: item-list/src/main/kotlin/com/gurunars/item_list/EmptyViewBinder.kt
package com.gurunars.item_list

import android.content.Context
import android.view.Gravity
import android.view.View
import android.widget.TextView
import com.gurunars.box.ui.fullSize

/** View binder for the case when there are no item in the list. */
typealias EmptyViewBinder = () -> View

/***/
fun Context.defaultEmptyViewBinder() = TextView(this).apply {
    id = R.id.noItemsLabel
    fullSize()
    text = getString(R.string.empty)
    gravity = Gravity.CENTER
}
